File: 645-set-mismatch/set-mismatch.java
import java.util.HashSet;

class Solution {
    public int[] findErrorNums(int[] nums) {
        int n = nums.length;
        int duplicate = 0;
        int missing = 0;
        HashSet<Integer> seen = new HashSet<>();
        for (int num : nums) {
            if (seen.contains(num)) {
                duplicate = num;
            }
            seen.add(num);
        }
        for (int i = 1; i <= n; i++) {
            if (!seen.contains(i)) {
                missing = i;
            }
        }
        return new int[]{duplicate, missing};
    }
}

File: 645-set-mismatch/set-mismatch.py
class Solution(object):
    def findErrorNums(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        n = len(nums)
        duplicate = 0
        missing = 0
        seen = set()
        for num in nums:
            if num in seen:
                duplicate = num
            seen.add(num)
        for i in range(1, n + 1):
            if i not in seen:
                missing = i
        return [duplicate, missing]